service/history/api/startworkflow/api.go

```diff
@@ -531,6 +535,9 @@ func (s *Starter) generateResponse(
 	workflowTaskInfo *workflow.WorkflowTaskInfo,
 	historyEvents []*historypb.HistoryEvent,
 ) (*historyservice.StartWorkflowExecutionResponse, error) {
+	if err := api.ProcessOutgoingSearchAttributes(s.shardContext, historyEvents, s.namespace.ID(), s.visibilityManager); err != nil {
```

Comment:
maybe move this below line 546 (`if !s.requestEagerStart()`) since it's not needed in the non-eager case?

Reply:
Oh good call, thanks